### PR TITLE
Add TimeTracker

### PR DESCRIPTION
**Pull request by ShaDy03**

### CODE REVIEW

File: IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using DataAccess.Models;
+
+namespace DataAccess.Data.IData;
+public interface ITimeTrackerData
+{
+    Task Add(TimeTracker entity);
+    Task Delete(int id);
+    Task<IEnumerable<TimeTracker>> GetAll();
```
Comment (iiusan) [inline IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs]: not a valid use case

File: IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿using DataAccess.Models;
+
+namespace DataAccess.Data.IData;
+public interface ITimeTrackerData
+{
+    Task Add(TimeTracker entity);
```
Comment (iiusan) [inline IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs]: missing the 'Async' suffix

File: IssueTracker/DataAccess/Models/TimeTracker.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿namespace DataAccess.Models;
+public class TimeTracker
+{
+    public string Name { get; set; }
+    public string Description { get; set; }
+    public DateTime Date { get; set; }
+    public int Worked { get; set; }
```
Comment (iiusan) [inline IssueTracker/DataAccess/Models/TimeTracker.cs]: What time unit are you using?

File: IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    [HttpPost("add-timeTracker")]
+    public async Task<IActionResult> Add(TimeTrackerRequest entity)
+    {
+        var timeTracker = _map.Map<TimeTracker>(entity);
+        await _trackerData.Add(timeTracker);
+        return Ok();
+    }
+
+    [HttpGet("getAll-timeTracker")]
+    public async Task<IActionResult> GetAll()
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: not a valid use case

Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: Time tracking details should be part of the issue response.

**Review by iiusan — CHANGES_REQUESTED** [review]

File: IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs (hunk 2 above)
Comment (ShaDy03) [reply IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs]: image

File: IssueTracker/DataAccess/Models/TimeTracker.cs (hunk 3 above)
Comment (ShaDy03) [reply IssueTracker/DataAccess/Models/TimeTracker.cs]: Datetime

File: IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs (hunk 4 above)
Comment (ShaDy03) [reply IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: I removed this method

File: IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs (hunk 1 above)
Comment (ShaDy03) [reply IssueTracker/DataAccess/Data/IData/ITimeTrackerData.cs]: In this file, I renamed all methods with the async suffix and I removed Getall method.

File: IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs (hunk 4 above)
Comment (ShaDy03) [reply IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: This method was removed.

File: IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs [hunk 5]
```diff
@@ -20,14 +21,20 @@ public TimeTrackerController(ITimeTrackerData trackerData)
     [HttpPost("add-timeTracker")]
     public async Task<IActionResult> Add(TimeTrackerRequest entity)
     {
+        if (!TimeTrackerValidator.IsValid(entity))
+            return BadRequest("Invalid TimeTracker!");
         var timeTracker = _map.Map<TimeTracker>(entity);
+        if (!TimeTrackerValidator.IsValid(timeTracker))
```
Comment (iiusan) [inline IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: Why is the request tested twice? You should validate the configuration inside the validation helper: https://docs.automapper.org/en/stable/Configuration-validation.html

File: IssueTracker/DataAccess/Data/TimeTrackerData.cs [hunk 6]
```diff
@@ -12,21 +12,30 @@ public TimeTrackerData(ISQLDataAccess db)
     }
     public async Task AddAsync(TimeTracker entity)
     {
+        if (String.IsNullOrEmpty(entity.Name))
```
Comment (iiusan) [inline IssueTracker/DataAccess/Data/TimeTrackerData.cs]: Create a validation method.

Comment (ShaDy03) [reply IssueTracker/DataAccess/Data/TimeTrackerData.cs]: I created this method

File: IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs (hunk 5 above)
Comment (ShaDy03) [reply IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: I removed this line. The mapping is correct.

Comment (iiusan) [reply IssueTracker/IssueTrackerAPI/Controllers/TimeTrackerController.cs]: Also, add the configuration validator for automaper.

**Review by iiusan — APPROVED** [review]

**Pull request merged**